base/strings/util.jl

```diff
@@ -358,7 +358,10 @@ _replace(io, repl, str, r, pattern) = print(io, repl)
 _replace(io, repl::Function, str, r, pattern) =
     print(io, repl(SubString(str, first(r), last(r))))
 
-function replace(str::String, pattern, repl, limit::Integer)
+function replace_new(str::String, pattern, repl, count::Int)
```

Comment:
It's what's done in other contexts too: accept `Integer` at the API level, but convert to `Int` in the method which does the work, a readons could be to limit the number of compiled functions. But I can see that there is no reason to error out if e.g. a very big `BigInt` is passed. I could use  `clamp(n, 0, typemax(Int)` instead of `Int(n)` in the caller (which is in deprecated now), assuming it's never necessary to replace more than `typemax(Int)` times.